### ISSUE TITLE
Disable OPA telemetry

### ISSUE THREAD

**Issue by sbernauer** [stackabletech/opa-operator#489]
We want to set the following flag for our OPA instances, as users can not (feasible) override the command and we have customers that certainly want to have it disabled.
`--disable-telemetry                    disables anonymous information reporting (see: https://www.openpolicyagent.org/docs/latest/privacy)`

**Comment by NickLarsenNZ** [comment]
Should that be an opt-in thing? I can imagine most want to disable telemetry, but perhaps some want to enable it?

**Comment by sbernauer** [comment]
Yeah, it should be an opt-in thing. We would need to add a CRD field and read that I guess. Changing the Issue to reflect that

**Comment by lfrancke** [comment]
I think we should just disable by default and have people use a config override for that. 

Is that possible in this case?

**Comment by sbernauer** [comment]
That's exactly what this Ticket wants. It's currently not possible to activate it, as we hard-code the start command

**Comment by lfrancke** [comment]
Sorry, I will clarify.

>  We would need to add a CRD field and read that I guess. 

I think we should _not_ add a CRD field for this _if_ it is possible to set this using a config override.

**Comment by sbernauer** [comment]
Ah I see. https://www.openpolicyagent.org/docs/latest/privacy/ only mentions the CLI flag, but there might also be a config setting.

**Comment by NickLarsenNZ** [comment]
Yeah, I dug through code and it doesn't look like there is a companion ENV var for it.

So if we explicitly disable it, and if someone really wants to enable, would they have to override the whole command? Or would we make an env var they can set to enable it and then adjust the command line based on it?

**Comment by sbernauer** [comment]
https://github.com/stackabletech/opa-operator/blob/d3ad0da6ad4178b2ab79f449091a9fa2cbb000ab/rust/operator-binary/src/controller.rs#L864C15-L864C15 adds the parameter unconditional to the start command. You could (in theory) overwrite it with podOverrides but that's pain in the ***.
IMHO we should add a field such as `spec.clusterConfig.telemetryEnabled` which defaults to false and add `--disable-telemetry` conditionally.

**Comment by lfrancke** [comment]
Ah! So do I understand this correctly that telemetry is already disabled now?

The [PR](https://github.com/stackabletech/opa-operator/pull/487/files) says "Closes" this one but this one is not closed.

I assume this one is now about making it possible to enable telemetry again?

**Comment by sbernauer** [comment]
> Ah! So do I understand this correctly that telemetry is already disabled now?

Yes, you are right. Originally this issues was "Disable telemetry", PR did so and closed this Issue.

Than @NickLarsenNZ had the request, that users should be able to opt in, and we re-opened the ticket changed it to reflect the new requirements

**Comment by lfrancke** [comment]
Thank you!
